Add access limits for projects in secrethub

hamir-suspect opened the pull request.

radwo reviewed (changes requested):
Why are we not using the same pattern as in the internal API? It's not that easy to change the public api.

hamir-suspect commented:
This was made a long time ago I think before we changed internal API, I just rebased it on monday, I'll revisit this once I'm done with `debug project api`

hamir-suspect commented:
#33 new version of this change
